gravitee-apim-console-webui/src/management/settings/api-portal-header/migrated/api-portal-header.component.ts

```diff
@@ -0,0 +1,247 @@
+}
+
+
+@Component({
+  selector: 'app-api-portal-header',
+  templateUrl: './api-portal-header.component.html',
+  styleUrls: ['./api-portal-header.component.scss']
+})
+export class ApiPortalHeaderComponent implements OnInit, OnDestroy {
+  protected readonly ToggleOptions = ToggleOptions;
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  public settings: PortalSettings = {};
+  public headersList: ApiPortalHeader[] = [];
+
+  displayedColumns: string[] = ['name', 'value', 'actions'];
+
+  constructor(
+    public environmentApiHeadersService: EnvironmentApiHeadersService,
+    private matDialog: MatDialog,
+    private snackBarService: SnackBarService,
+    private portalSettingsService: PortalSettingsService,
+    private readonly permissionService: GioPermissionService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.initListSubscription();
+    this.getList();
+    this.initSettings();
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  private initSettings(): void {
+    this.portalSettingsService
+      .get()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((settings: PortalSettings) => {
+        this.settings = settings;
+      });
+  }
+
+  private initListSubscription() {
+    this.environmentApiHeadersService
+      .getHeadersList$()
+      .pipe(
+        catchError(_ => {
+          this.snackBarService.error('Error occurred.');
+          return [];
+        }),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe((list: ApiPortalHeader[]): void => {
+        this.headersList = list;
+      });
+  }
+
+  private createMessage(option: ToggleOptions): string {
+    const messages = {
+      [ToggleOptions.show_tags]: 'Tags are now ' + (this.settings.portal.apis.apiHeaderShowTags.enabled ? 'visible' : 'hidden'),
+      [ToggleOptions.show_categories]: 'Categories are now ' + (this.settings.portal.apis.apiHeaderShowCategories.enabled ? 'visible' : 'hidden'),
+      [ToggleOptions.promoted_api_mode]: 'Promoted API is now ' + (this.settings.portal.apis.promotedApiMode.enabled ? 'visible' : 'hidden')
+    };
+    return messages[option];
+  }
+
+
+  public saveToggle(toggleOption: ToggleOptions): void {
+    this.portalSettingsService.get()
+      .pipe(
+        switchMap((currentSettings: PortalSettings) => {
+          return this.portalSettingsService.save(
+            { ...currentSettings, portal: { ...currentSettings.portal, apis: this.settings.portal.apis } });
+        }),
+        tap(() => {
+          this.snackBarService.success(this.createMessage(toggleOption));
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe((updatedSettings: PortalSettings) => {
+        this.settings = updatedSettings;
+      });
+  };
+
+  public isReadonlySetting(property: string): boolean {
```

Comment:
I think there's a bit missing about use this function. 
you can fond some Jan code for this [here](https://github.com/gravitee-io/gravitee-api-management/pull/6708/files#diff-1a551c6b6fbc9f8716e7a48cc95bdca121714e9634f336289780a5bf57c116a5)

basically, if this configuration is set in a Yaml on the back end, it cannot be modified here. That why we have this kind of 🔒 on all settings here

Reply:
OK, checking the status now in two functions:[disabled]="isReadonlySetting('portal.apis.apiheader.showcategories.enabled') || !canUpdateSettings()"

  public isReadonlySetting(property: string): boolean {
    return PortalSettingsService.isReadonly(this.settings, property);
  };

  public canUpdateSettings(): boolean {
    return this.permissionService.hasAnyMatching([
      'environment-settings-c',
      'environment-settings-u',
      'environment-settings-d',
    ]);
  }